Mapping/ContainerEntityListenerResolver.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace Doctrine\Bundle\DoctrineBundle\Mapping;
+
+use InvalidArgumentException;
+use Psr\Container\ContainerInterface;
+use RuntimeException;
+
+class ContainerEntityListenerResolver implements EntityListenerServiceResolver
```

Comment:
Should this class be `final`? Since that packaged moved from the sf namespace to Doctrine it would make sense to apply that kind of policy to new types / methods, right?

Reply:
Currently it cannot be final as the old/legacy class `ContainerAwareEntityListenerResolver` extends this new class.

Reply:
maybe we can add `@final` for now and add the keyword in version2? @alcaeus ?

Reply:
Yes, `@final` works for now. It would mean that people using the old class would probably two warnings from the debug class loader (deprecation of the old class as well as warning because the new class would be final), but I'm hoping it will just get them to migrate away faster ;)